artifacttemplates/http%3A%2F%2Fopentosca.org%2Fartifacttemplates/OpenStack_CloudProviderInterfaceIA-w2/source/src/main/java/org/opentosca/artifacttemplates/WsdlFilter.java

```diff
@@ -0,0 +1,55 @@
+package org.opentosca.artifacttemplates;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class WsdlFilter implements Filter {
```

Comment:
Isn't this WsdlFilter really needed? It looks a bit overengineered

Reply:
No, because spring does not support to get the wsdl using a query parameter. This adds the support for this by routing `?wsdl` to `.wsdl`.